datamol/io.py

```diff
@@ -257,3 +321,37 @@ def read_smi(
         pathlib.Path(str(active_path)).unlink()
 
     return mols
+
+
+def to_xlsx(
+    mols: Union[Mol, Sequence[Mol], pd.DataFrame],
+    urlpath: Union[str, os.PathLike],
+    smiles_column: Optional[str] = "smiles",
+    mol_column: str = "mol",
+    mol_size: List[int] = [300, 300],
+):
+    """Write molecules to an Excel file with a molecule column as an RDKit rendered
+    image.
+
+    Args:
+        mols: a dataframe, a molecule or a list of molecule.
+        urlpath: Path to a file or a file-like object. Path can be remote or local.
+        smiles_column: Column name to extract the molecule.
+        mol_column: Column name to extract the molecule. It takes
+            precedence over `smiles_column`.
+            Column name to write the RDKit rendered image. If none,
+            the molecule images are not written.
+    """
+
+    if isinstance(mols, Mol):
+        mols = [mols]
+
+    if isinstance(mols, list):
```

Comment:
this check does not conform with the typing.Sequence instance expected as input. 

```python
from typing import Sequence
t = tuple(range(2))
isinstance(t, Sequence)#  == True
```

either change the input type or test something broader here.

Reply:
Thanks, I changed it to `Sequence`.

Honestly, I never know what to do for list typing. I am often looking for a sequence like types with defined length and ideally compatible with numpy 1d array.

What do you do usually?

Reply:
If you noticed other inconsistent types in datamol, please let me know (or open a ticket).

Reply:
I don't think there is a generic good answer for the right typing here, unless a specific type object is defined (see for example what numpy does with:`numpy.typing.ArrayLike`